app/src/main/java/org/oppia/android/app/survey/surveyitemviewmodel/MarketFitItemsViewModel.kt

```diff
@@ -0,0 +1,127 @@
+package org.oppia.android.app.survey.surveyitemviewmodel
+
+import androidx.databinding.Observable
+import androidx.databinding.ObservableArrayList
+import androidx.databinding.ObservableField
+import androidx.databinding.ObservableList
+import org.oppia.android.R
+import org.oppia.android.app.model.MarketFitAnswer
+import org.oppia.android.app.model.SurveyQuestionName
+import org.oppia.android.app.model.SurveySelectedAnswer
+import org.oppia.android.app.survey.SelectedAnswerAvailabilityReceiver
+import org.oppia.android.app.survey.SelectedAnswerHandler
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import javax.inject.Inject
+
+/** [SurveyAnswerItemViewModel] for the market fit question options. */
+class MarketFitItemsViewModel @Inject constructor(
+  private val resourceHandler: AppLanguageResourceHandler,
+  private val selectedAnswerAvailabilityReceiver: SelectedAnswerAvailabilityReceiver,
+  private val answerHandler: SelectedAnswerHandler
+) : SurveyAnswerItemViewModel(ViewType.MARKET_FIT_OPTIONS) {
+  val optionItems: ObservableList<MultipleChoiceOptionContentViewModel> = getMarketFitOptions()
+
+  private val selectedItems: MutableList<Int> = mutableListOf()
+
+  override fun updateSelection(itemIndex: Int, isCurrentlySelected: Boolean): Boolean {
+    optionItems.forEach { item -> item.isAnswerSelected.set(false) }
+    if (!selectedItems.contains(itemIndex)) {
+      selectedItems.clear()
+      selectedItems += itemIndex
+    } else {
+      selectedItems.clear()
+    }
+    updateIsAnswerAvailable()
+    return selectedItems.isNotEmpty()
+  }
+
+  val isAnswerAvailable = ObservableField(false)
+
+  init {
+    val callback: Observable.OnPropertyChangedCallback =
+      object : Observable.OnPropertyChangedCallback() {
+        override fun onPropertyChanged(sender: Observable, propertyId: Int) {
+          selectedAnswerAvailabilityReceiver.onPendingAnswerAvailabilityCheck(
+            selectedItems.isNotEmpty()
+          )
+          getPendingAnswer()
+        }
+      }
+    isAnswerAvailable.addOnPropertyChangedCallback(callback)
+  }
+
+  private fun updateIsAnswerAvailable() {
+    val wasSelectedItemListEmpty = isAnswerAvailable.get()
+    if (selectedItems.isNotEmpty() != wasSelectedItemListEmpty) {
```

Comment:
If this is a boolean, I would suggest not starting it with a verb -- i.e. selectedItemListWasEmpty

Reply:
Thanks, I have updated this.